addons/pos_loyalty/static/src/js/Loyalty.js

```diff
@@ -1632,12 +1631,20 @@ patch(Order.prototype, "pos_loyalty.Order", {
                 this._updateRewards();
             }
         }
+        if (!rule) {
```

Comment:
I think the more correct condition is:
1. Is the coupon in the codeActivatedCoupons?
2. Are there orderlines?

Reply:
Why checking for orderlines ? and if the coupon was not in the codeActivatedCoupons it would not reach that part of the code

Reply:
> Why checking for orderlines ?

I believe we only want to show the notification if the cart is empty which is the same to having no orderlines.

> and if the coupon was not in the codeActivatedCoupons it would not reach that part of the code

I guess then the proper solution is to make an 'rpc' asking the server the balance of the scanned gift card.